.github/templates/wrappers.yml

```diff
@@ -86,15 +86,20 @@ jobs:
     runs-on: macos-12
     name: macOS
     _: #@ template.replace(buildWrappers("./wrappers/build-macos.sh", "wrappers-macos"))
-  ios:
+  catalyst:
     runs-on: macos-12
-    name: iOS
+    name: Catalyst
+    _: #@ template.replace(buildWrappers("pwsh ./wrappers/build-apple-platform.ps1 Catalyst", "wrappers-catalyst"))
+  apple-platforms:
+    runs-on: macos-12
+    name: Apple Platform
     strategy:
       matrix:
-        arch: #@ iOSArchs
-    _: #@ template.replace(buildWrappers("pwsh ./wrappers/build-ios.ps1 ${{ matrix.arch }}", "wrappers-ios-${{ matrix.arch }}", enableLto = False, intermediateSteps = [setupXcode()]))
+        platform: #@ applePlatforms
+        target: [ 'Device', 'Simulator' ]
+    _: #@ template.replace(buildWrappers("pwsh ./wrappers/build-apple-platform.ps1 ${{ matrix.platform }} -Targets ${{ matrix.target }}", "wrappers-apple-${{ matrix.platform }}-${{ matrix.target }}", enableLto = False, intermediateSteps = [setupXcode()]))
     #@yaml/map-key-override
-    if: #@ " || ".join([ "needs.check-cache.outputs.wrappers-ios-" + x + " != 'true'" for x in iOSArchs ])
+    if: #@ " || ".join([ "needs.check-cache.outputs.wrappers-apple-" + x + "-Device != 'true'" for x in applePlatforms ])
```

Comment:
[nit] Do we need the `-apple` here? Seems like this is producing artifacts like `wrappers-apple-ios-device`, which is kind of verbose.